src/test/java/ResponseProcessorTest.java

```diff
@@ -0,0 +1,65 @@
+import com.google.common.collect.Lists;
+import eventprocessing.models.Message;
+import eventprocessing.models.Response;
+import eventprocessing.models.Sensor;
+import eventprocessing.models.SensorList;
+import eventprocessing.responseservices.ResponseProcessor;
+import eventprocessing.storage.MessageLog;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.mockito.Mockito;
+
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+import static org.mockito.Mockito.when;
+
+public class ResponseProcessorTest {
+
+    private Sensor mockSensor;
+    private SensorList mockSensorList;
+    private Message mockMessage;
+    private Response mockResponse;
+    private MessageLog mockMessageLog;
+    private ResponseProcessor responseProcessor;
+
+    @Before
+    public void setup() {
+        mockSensor = Mockito.mock(Sensor.class);

```

Comment:
This long setup in `@Before` makes it hard to follow the individual tests. For example, if you compare the tests on line 43 and line 49, it's not obvious _why_ `validId` returns `true`, and `invalidId` returns `false`.

Lots of shared setup also leads to fragile tests - you tweak some setup to add a new test, and it breaks an existing one. Or (worse) an existing one now passes for the wrong reason.

Try to put all the setup for a test within the test itself. Sometimes you can't avoid it, because the tests would be too slow if they did their own setup, but that normally only applies to integration tests that are slow because they set up database fixtures.

You can create helper methods like builders to simplify test setup, and you can design those so that the test setup is still obvious (i.e. without hiding data).

If your tests are still too long, it's usually a sign that the code under test has a complicated interface, and it should be simplified.